agent.go

```diff
@@ -1210,3 +1222,51 @@ func winsizeCb(pod *pod, data []byte) error {
 
 	return nil
 }
+
+// Get the agent configuration from kernel cmdline
+func getConfig() error {
+	kernelCmdline, err := ioutil.ReadFile(kernelCmdlineFile)
+	if err != nil {
+		return err
+	}
+	if err := parseKernelCmdline(string(kernelCmdline)); err != nil {
+		return err
+	}
+	return nil
+}
+
+//Parse a string that represents a kernel cmdline to get agent options
+func parseKernelCmdline(cmdline string) error {
+
+	const valuePosition = 1
+	const optionSeparator = "="
+
+	words := strings.Fields(cmdline)
+
+	for _, w := range words {
+		for o := range agentOptions {
```

Comment:
I am sorry but I don't like that we loop on agentOptions and at the same time we are modifying its values. Try something  more simple to stay as clear as possible:
I think we should expect `getConfig()` to return a config, and for each possible option parsed by `parseKernelCmdLine()`, we want a default in case it is not provided by the command line.
This means that `parseKernelCmdLine()` will build a list of options, that is returned to `getConfig()`, and from the `main()`, we apply this list of options in any case because we know that we will always have some default values (a function like `applyConfig()` would be fine). No need to specify any error, maybe just a `log.Warn()` in case we could read `/proc/cmdline` for example.

Moreover, about the way you loop over every element of this, because this is a map, you can directly have `key/value` from the loop for:
```go
for key, value := range optionList {
    ...
}
```